storm-core/src/jvm/backtype/storm/scheduler/resource/SchedulingResult.java

```diff
@@ -0,0 +1,111 @@
+
+    //contains the result for the attempted scheduling
+    private Map<WorkerSlot, Collection<ExecutorDetails>> schedulingResultMap = null;
+
+    private SchedulingStatus status = null;
+
+    private String message = null;
+
+    private String errorMessage = null;
+
+    private static final Logger LOG = LoggerFactory.getLogger(SchedulingResult.class);
+
+
+    public SchedulingResult(SchedulingStatus status, Map<WorkerSlot, Collection<ExecutorDetails>> schedulingResultMap, String message, String errorMessage) {
+        this.status = status;
+        this.schedulingResultMap = schedulingResultMap;
+        this.message = message;
+        this.errorMessage = errorMessage;
+    }
+
+    public static SchedulingResult failure(SchedulingStatus status, String errorMessage) {
+        return new SchedulingResult(status, null, null, errorMessage);
+    }
+
+    public static SchedulingResult success(Map<WorkerSlot, Collection<ExecutorDetails>> schedulingResultMap) {
+        return SchedulingResult.successWithMsg(schedulingResultMap, null);
+    }
+
+    public static SchedulingResult successWithMsg(Map<WorkerSlot, Collection<ExecutorDetails>> schedulingResultMap, String message) {
+        if (schedulingResultMap == null) {
+            throw new IllegalStateException("Cannot declare scheduling success without providing a non null scheduling map!");
+        }
+        return new SchedulingResult(SchedulingStatus.SUCCESS, schedulingResultMap, message, null);
+    }
+
+    public SchedulingStatus getStatus() {
+        return this.status;
+    }
+
+    public String getMessage() {
+        return this.message;
+    }
+
+    public String getErrorMessage() {
+        return this.errorMessage;
+    }
+
+    public Map<WorkerSlot, Collection<ExecutorDetails>> getSchedulingResultMap() {
+        return schedulingResultMap;
+    }
+
+    public boolean isSuccess() {
+        return SchedulingStatus.isStatusSuccess(this.status);
+    }
+
+    public boolean isFailure() {
+        return SchedulingStatus.isStatusFailure(this.status);
+    }
+
+    public boolean isValid() {
+        if (this.isSuccess() && this.getSchedulingResultMap() == null) {
+            LOG.warn("SchedulingResult not Valid! Status is success but SchedulingResultMap is null");
+            return false;
+        }
+        if (this.isFailure() && this.getSchedulingResultMap() != null) {
+            LOG.warn("SchedulingResult not Valid! Status is Failure but SchedulingResultMap is NOT null");
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public String toString() {
+        String ret = "";
+        if(this.isSuccess()) {
+            ret += "Status: " + this.getStatus() + " message: " + this.getMessage() + " scheduling: " + this.getSchedulingResultMap().toString();
+        } else {
+            ret += "Status: " + this.getStatus() + " error message: " + this.getErrorMessage();
+        }
+        return ret;
```

Comment:
If we dropped the += it would still work and ret could just be set to null to start out with.